oneargopy/Argo.py

```diff
@@ -1019,3 +1152,300 @@ def __determine_graph_step(self, ax, axis: str)-> int:
+
+
+    def __parameter_premutations(self, nc_file)-> list :
+        """ A function to filter the list of parameters to be loaded so 
+            that we only load parameters that are in the file.  
+
+            :param: parameters : list - A list of parameters to include
+                in the dataframe.
+            :param: nc_file : Any - The .nc file we're reading from. 
+
+            :return: list - A list to of all the parameters passed
+                that are inside of the nc_file.
+        """
+        # If the parameter is in the file also add it's permutations to the list
+        if isinstance(self.float_parameters, list) :
+
+            # Parameters that are in the passed .nc file
+            file_variables = nc_file.variables
+
+            # List to store parameters and their additioal associated columns
+            parameter_columns = []
+            
+            for parameter in self.float_parameters : 
+                if parameter in file_variables: 
+                    # We add PRES no matter what, so if the user passed it 
+                    # don't add it to the parameter list at this time.
+                    if parameter != 'PRES' : 
+                        parameter_columns.append(parameter)
+                        parameter_columns.append(parameter + '_QC')
+                        parameter_columns.append(parameter + '_ADJUSTED')
+                        parameter_columns.append(parameter + '_ADJUSTED_QC')
+                        parameter_columns.append(parameter + '_ADJUSTED_ERROR')
+                else : 
+                    print(f'WARNING: The parameter: {parameter} does not exist in the file.')
+            
+            if len(parameter_columns) > 0 : 
+                pressure = ['PRES', 'PRES_QC', 'PRES_ADJUSTED', 'PRES_ADJUSTED_QC', 'PRES_ADJUSTED_ERROR']
+                existing_parameter_columns = pressure + parameter_columns
+                return existing_parameter_columns
+            else : 
+                return None
+        
+        else : 
+            return None
+
+    
+    def __fill_float_data_dataframe(self, files: list)-> pd: 
+        """ A Function to load data into the float data dataframe.
+ 
+            :param: files : list - A list of files to read in data from.
+
+            :return: pd : Dataframe - The dataframe of float data with rows 
+                where measurements were not collected excluded.
+        """
+        if self.download_settings.verbose: print(f'Loading float data...')
+
+        # Getting the file paths for downloaded .nc files
+        directory = Path(self.download_settings.base_dir.joinpath("Profiles"))
+        file_paths = []
+        for file in files : 
+            file_paths.append(directory.joinpath(file))
+
+        # Columns that will always be in the dataframe, these columns are one dimensional
+        static_columns = ['WMOID', 'CYCLE_NUMBER', 'DIRECTION', 
+                                'DATE', 'DATE_QC', 'LATITUDE', 
```

Comment:
all _QC variables should appear as integer numbers in the csv file, not 
b'1'
but
1

Reply:
Okay so, I had to change how i was handling decoding the masked arrays / byte arrays because even though they have the same formatting they're considered different datatypes. You had added something to my log sheet: 
`doxy_qc.data[doxy_qc.data == b' '] = b'0'`
`decoded_unmasked = np.vectorize(lambda x: int(x.decode('utf-8')))(doxy_qc.compressed())`
but I couldn't integrate this in a way that worked. 
This is what i settled on: 

![image](https://github.com/user-attachments/assets/0ea32c3a-9c21-436f-936d-7707ce3254b0)

![image](https://github.com/lovelysandlonelys/OneArgo-Python-Dev/assets/91633941/ada860db-5413-4a9f-9c9b-018108f6d6ff)